src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -30,6 +30,8 @@ export interface DatasetTypeConfig {
     icon: EuiIconProps;
     /** Optional tooltip text */
     tooltip?: string;
+    /** Optional preference for search on page load else defaulted to true */
+    searchOnLoad?: boolean;
```

Comment:
nit: So you set this one as an optional property. Seems searchOnLoad property is added to some dataset types but not all? Seems we have searchOnLoad on index pattern type, index type and s3 type. what else types need this searchOnLoad that allows this property to be missing? To make it consistent, maybe remove ? because this `return !datasetType || (datasetService?.getType(datasetType)?.meta?.searchOnLoad ?? true` already set the default value to true right? so `?` is not necessary.

Reply:
nice callout. I think we should have different types for the config we provide and the config we receive from the service. Not a blocker for the PR though